pandas/core/window/rolling.py

```diff
@@ -2246,15 +2295,35 @@ def _create_blocks(self, obj: FrameOrSeriesUnion):
         obj = obj.take(groupby_order)
         return super()._create_blocks(obj)
 
-    def _get_cython_func_type(self, func: str) -> Callable:
+    def calculate_center_offset(self, window, center) -> int:
+        """
+        Calculate an offset necessary to have the window label to be centered.
+
+        Parameters
+        ----------
+        window: ndarray or int
+            window weights or window
+
+        Returns
+        -------
+        int
+        """
+        if not center or not self.win_type:
+            return 0
+
+        if not is_integer(window):
```

Comment:
can we not just make this a free function? i get that you are passing win_type here, but that could easily be passed in as an arg

Reply:
I again tried making this a free function and confirmed that the two classes require different functionality (whether or not to include `or not self.win_type` in the if statement).

Reply:
You can just add an optional 3rd kwarg `(window, center: bool, win_type: Optional[str] = None)`

Reply:
I did try that and it did not work. The issue is that in one class win_type should be completely ignored whereas the other class needs to use it. But both classes share the `_apply` function, which actually calls `calculate_center_offset` [here](https://github.com/pandas-dev/pandas/pull/36132/files#diff-0de5c5d9abfcdd141e83701eaaec4358R566-R569).

So I can't just do `calculate_center_offset(..., None)` for the first class and `calculate_center_offset(..., self.win_type)` for the second (because there's only one shared call to `calculate_center_offset(...)`, not two separate ones)

Reply:
ok see my comment above; if this is now a method, then you can simply use is_freq_type and this PR is a lot simpler.

Reply:
@jreback Are you cool with this implementation?

Reply:
I don't understand the skip_offset parameter. why is it not sufficient to just make a property on the class itself (e.g. you can make another preoprty / method if needed, similr to is_ffreq_type). passing parameters around like this is *really* hard to understand.

Reply:
@justinessert I think same as above, can you explain why skip_offset arg is needed